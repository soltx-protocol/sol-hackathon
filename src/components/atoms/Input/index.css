.input {
	position: relative;

	& > input {
		width: 100%;
		height: 48px;
		padding: 9px 13px;
		border-radius: 6px;
		background-color: var(--neutral-white);
		border: solid 1px var(--divider-20);
		transition: border 0.3s ease-out;
		font-style: normal;
		font-weight: 400;
		font-size: 16px;
		line-height: 24px;
		text-align: left;
		color: var(--text);
		box-shadow: 0 1px 2px rgb(0 0 0 / 5%);

		&::placeholder {
			font-style: normal;
			font-weight: 400;
			font-size: 16px;
			line-height: 24px;
			text-align: left;
			color: var(--text-50);
		}

		&:hover,
		&:focus {
			border-color: var(--primary-cta);
		}
	}

	&.icon {
		& > input {
			padding-right: 40px;
		}

		& > .icon {
			display: flex;
			justify-content: center;
			align-items: center;
			position: absolute;
			right: 6px;
			top: 12px;
			width: 24px;
			height: 24px;

			& > svg {
				width: 24px;
				height: 24px;
			}
		}

		&.is-left-icon {
			& > input {
				padding-left: 40px;
				padding-right: 0;
			}

			& > .icon {
				right: unset;
				left: 6px;
				width: 24px;

				& > svg {
					width: 24px;
					height: 24px;
				}
			}
		}
	}

	/* stylelint-disable-next-line no-descending-specificity */
	&.error > input {
		border-color: var(--system-red);
	}
}
