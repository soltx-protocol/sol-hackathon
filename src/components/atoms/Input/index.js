/* eslint-disable react/jsx-props-no-spreading */

import React from 'react';
import classnames from 'classnames';

// import EyeIcon from 'images/icon/eye-show.inline.svg';
// import EyeHideIcon from 'images/icon/eye-hide.inline.svg';

import { isExist } from 'util/helper';

import styles from './index.css';

const Input = ({
	className,
	Icon,
	isLeftIcon,
	onClickIcon = () => {},
	disabled = false,
	error = false,
	...other
}) => (
	<div
		className={classnames(styles.input, className, {
			[styles.icon]: isExist(Icon),
			[styles.isLeftIcon]: isExist(Icon) && isLeftIcon,
			[styles.disabled]: disabled,
			[styles.error]: error,
		})}
	>
		<input disabled={disabled} {...other} />
		{isExist(Icon) && (
			<button className={styles.icon} onClick={onClickIcon} type="button">
				<Icon />
			</button>
		)}
	</div>
);

export default Input;
