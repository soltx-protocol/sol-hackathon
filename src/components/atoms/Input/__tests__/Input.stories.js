import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean } from '@storybook/addon-knobs/react';

import Input, { InputSearch } from 'components/atoms/Input';

const stories = storiesOf('COMMON|atoms/Input', module);

stories.add('__interactive', () => (
	<Input
		disabled={boolean('disabled', false)}
		error={boolean('error', false)}
		placeholder="請輸入"
	/>
));

stories.add('with search input', () => <InputSearch placeholder="請輸入" />);
