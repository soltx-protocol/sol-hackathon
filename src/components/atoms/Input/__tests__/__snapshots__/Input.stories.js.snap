// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    class="input"
  >
    <input
      placeholder="請輸入"
    />
  </div>
</DocumentFragment>
`;
